s3fs/core.py

```diff
@@ -943,6 +944,12 @@ async def _put_file(self, lpath, rpath, chunksize=50 * 2 ** 20, **kwargs):
             else:
                 await self._mkdir(lpath)
         size = os.path.getsize(lpath)
+
+        content_type, _ = mimetypes.guess_type(lpath)
+
+        if content_type is not None:
```

Comment:
I would change the logic around so that we don't do the lookup if the user gives a value, and not to bother setting a None:
```
if "ContentType" not in kwargs:
    content_type, _ = mimetypes.guess_type(lpath)
    if content_type is not None:
        kwargs["contentType"] = content_type
```

I assume that the implicit default is octet-stream, perhaps we should mention this in a comment or docs.

Reply:
good idea, i've added this in.